src/objectc.c

```diff
@@ -125,6 +125,45 @@ int msgpack_pack_object(msgpack_packer* pk, msgpack_object d)
     }
 }
 
+int msgpack_object_init(msgpack_object* d, void *data, size_t size, int type)
```

Comment:
```suggestion
bool msgpack_object_init(msgpack_object* d, int type, void* data, size_t size)
```

1. Please update the order of the parameter.
2. Please apply coding rule. `void *data` -> `void* data`.